lib/internal/fs/promises.js

```diff
@@ -417,6 +418,13 @@ async function ftruncate(handle, len = 0) {
   return binding.ftruncate(handle.fd, len, kUsePromises);
 }
 
+async function rm(path, options) {
+  path = pathModule.toNamespacedPath(getValidatedPath(path));
+  options = validateRmOptionsSync(path, options);
```

Comment:
I think we should use the async validation here, otherwise we're going to create a bottleneck when performing many `rm` operations, you can do something like this:

```js
options = await new Promise((resolve, reject) => {
  validateRmOptionsSync(path, options, false, (err, options) => {
     if (err) return reject(err);
     else return resolve(options);
  });
})
return rimrafPromises(path, options);
```

Reply:
isn't this still creating the same bottleneck tho, it's just deferring the result? a `new Promise` executor runs synchronously.

Reply:
@ljharb my bad, I meant:

```js
options = await new Promise((resolve, reject) => {
  validateRmOptions(path, options, false, (err, options) => {
     if (err) return reject(err);
     else return resolve(options);
  });
})
return rimrafPromises(path, options);
```

i.e., not using the `sync` version of the validation.

Reply:
I implemented this a few days ago and I did catch that mistake in your suggested change @bcoe. I'm not sure if that changes the point @ljharb raised.

Reply:
This hasn't been updated yet, but yes, @bcoe's updated suggestion would address my point. I'd also say that it's worth adding a custom promisify implementation to `validateRmOptions`, and then use `await promisify(validateRmOptions)(path, options, false)` instead?